src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -240,6 +246,11 @@ public int getInitOnlineTimeoutSec() {
         return initOnlineTimeoutSec == null ? DEFAULT_INIT_ONLINE_TIMEOUT_SEC : initOnlineTimeoutSec;
     }
 
+    public int getScheduledFutureTimeoutSec() {
+        // Wait 3 update cycles before timing out. Gives a little cushion in case fleet is under modification
+        return getCloudStatusIntervalSec() * 3;
```

Comment:
Cloud Status Interval could be as little as 1-3 seconds. Do you think we should be depending on cloudnanny interval or have sort of constant value of 60-90 seconds?

Reply:
I think it should be based on the Cloud Status Interval because we will only cancel timeout when update() is run, which depends on Cloud Status Interval. And if the interval is small then we might still get value from timing out early if another fleet could handle that job. 

I'm not against adding a default longer minimum though for cases when Cloud Status Interval is small. Maybe something like
```
private final int MIN_SCHEDULED_FUTURE_TIMEOUT = 30;

public int getScheduledFutureTimeoutSec() {
  final int timeout = getCloudStatusIntervalSec() * 3;
  return timeout > MIN_SCHEDULED_FUTURE_TIMEOUT ? timeout : MIN_SCHEDULED_FUTURE_TIMEOUT;
}
```

Thoughts?

Reply:
I was thinking about a scenario when a customer has launched way too many instances in their previous update and `fleet under modification` will last longer than ``cloud interval * 3`` (where cloud interval is small).  AFAIK, fleet will remain in modifying state until the target capacity is achieved. So having something like `MIN_SCHEDULED_FUTURE_TIMEOUT` might be helpful.

Reply:
For that case, if we scale up in the previous update the new scheduled futures will be cancelled and the Jenkins futures will be left alone. 

So it'd look like
1. Add jobs to queue
2. Jenkins finds jobs and calls provision on fleet
3. Scheduled futures start
4. Update happens, we start to scale up
5. Scheduled futures are cancelled
6. Next update is called, fleet is still scaling so it's in modification state now

Reply:
I meant the scenario where:

1. Add jobs to queue
2. Jenkins updates fleet target to a relatively large number
3. Add more jobs to queue
4. Target will not be updated as fleet is still in modifying state launching large number of instances
5. The second round of futures might timeout if Cloud status interval is small

Reply:
I think that's actually ok though, because if those futures are cancelled then another fleet might be able to provision capacity for those jobs